src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -240,6 +304,42 @@ private static int GetCurrentProcessId()
             return Interop.Sys.GetPid();
         }
 
+        /// <summary>
+        /// Returns all immediate child processes.
+        /// </summary>
+        private IReadOnlyList<Process> GetChildProcesses()
+        {
+            List<Process> childProcesses = new List<Process>();
+
+            foreach (Process possibleChildProcess in GetProcesses())
+            {
+                bool keep = false;
+
+                try
+                {
+                    if (SafePredicateTest(() => IsParentOf(possibleChildProcess)))
+                    {
+                        childProcesses.Add(possibleChildProcess);
+                        keep = true;
+                    }
+                }
+                finally
+                {
+                    if (!keep)
+                        possibleChildProcess.Dispose();
+                }
+            }
+
+            return childProcesses;
+        }
+
+        /// <summary>Checks whether the argument is a direct child of this process.</summary>
+        private bool IsParentOf(Process possibleChildProcess) =>
```

Comment:
@danmosemsft, is this something you'd really like done or just an idea? :-) 

I am glad to make this change if you want it done. If it's just an idea where you're okay with going either way on it, I wouldn't mind sticking with `IsParentOf` as that terminology matches what's used elsewhere in this PR (e.g. we ask the OS for the parent process ID/ppid, not the immediate parent process ID/ippid).

All this said, if you want the change made, just say so and it will be done.

Reply:
Fine to leave as is :)

Reply:
Thanks! Hope you have a great weekend!